### PR TITLE
fix #3

### LINKED ISSUE #3: Add testing!!
No testing reduces the quality of the code... let people know it works

### PR DESCRIPTION
**Pull request by ins0**
This PR fixes #3 

- [x] add phpunit test framework as `DEV` dependency
- [x] add first `Producer` Test Cases

### CODE REVIEW

**Comment by beyerz** [comment]
@ins0, Thank you for this pull request. Testing is extremely important to ensure that the code keeps working and that this bundle can be trusted.

I have checked out your branch for the pull request and run PHPUnit against it.
Unfortunately, tests are failing and are not actually related to your code
We can either merge your changes and then update the code to ensure that tests are passing, or we can fix this on your current branch, I would prefer to do this on the current branch as the fix is very minor.
Here are the fail reasons:
Beyerz\AWSQueueBundle\Fabric\GearmanFabric is not implementing the interface correctly
You can simply implement the interface without implementing any logic in the functions.

File: phpunit.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+         convertWarningsToExceptions="true"
+         processIsolation="false"
+         stopOnFailure="false">
+    <testsuites>
+        <testsuite name="AWS Queue Bundle Test Suite Unit">
+            <directory>Tests/Unit</directory>
+        </testsuite>
+    </testsuites>
+    <filter>
+        <whitelist processUncoveredFilesFromWhitelist="true">
```
Comment (beyerz) [inline phpunit.xml]: Please could you correct this as it gives a no coverage error as the directory 'src/' does not exist as well as add a blacklist.
I have added the code block below for you to make it easier for us as I already have tested with this :)

```xml
<filter>
        <blacklist>
            <directory suffix=".php">./vendor</directory>
        </blacklist>
        <whitelist processUncoveredFilesFromWhitelist="true">
            <directory suffix=".php">./</directory>
        </whitelist>
</filter>

**Comment by ins0** [comment]
hey @beyerz can u give me hint how i can reproduce your failing tests? not sure where i have to fix this interface problem. ty!

**Comment by beyerz** [comment]
@ins0 

If you go to this file: [Beyerz\AWSQueueBundle\Fabric\GearmanFabric](https://github.com/beyerz/AwsQueueBundle/blob/master/Fabric/GearmanFabric.php)

You will see this code:
```
<?php
/**
 * Created by PhpStorm.
 * User: beyerz
 * Date: 24/07/2018
 * Time: 14:41
 */

namespace Beyerz\AWSQueueBundle\Fabric;


class GearmanFabric extends AbstractFabric
{

    /**
     * Fabric should ensure that all notification channels and respective queues exist and subscribers are defined
     * @return mixed
     */
    public function setup()
    {
        // TODO: Implement setup() method.
    }
}
```
but the AbstractFabric Class requires certain methods to exist. so the class should look like this.
(You can copy paste from here is this should work)
```
<?php
/**
 * Created by PhpStorm.
 * User: beyerz
 * Date: 24/07/2018
 * Time: 14:41
 */

namespace Beyerz\AWSQueueBundle\Fabric;


use Beyerz\AWSQueueBundle\Consumer\ConsumerService;
use Doctrine\Common\Collections\ArrayCollection;

class GearmanFabric extends AbstractFabric
{


    /**
     * Fabric should ensure that all notification channels and respective queues exist and subscribers are defined
     * @param string          $channel
     * @param ArrayCollection $subscribers
     * @return mixed
     */
    public function setup(string $channel, ArrayCollection $subscribers)
    {
        // TODO: Implement setup() method.
    }

    /**
     * @param string $channel
     * @return mixed
     */
    public function setupNotificationServices(string $channel)
    {
        // TODO: Implement setupNotificationServices() method.
    }

    /**
     * @param string          $channel
     * @param ArrayCollection $subscribers
     * @return mixed
     */
    public function setupQueueServices(string $channel, ArrayCollection $subscribers)
    {
        // TODO: Implement setupQueueServices() method.
    }

    /**
     * @param string          $message
     * @param string          $channel
     * @param ArrayCollection $subscribers
     * @return mixed
     */
    public function publish(string $message, string $channel, ArrayCollection $subscribers)
    {
        // TODO: Implement publish() method.
    }

    /**
     * @param ConsumerService $consumer
     * @param int             $messageCount
     * @return int
     */
    public function consume(ConsumerService $consumer, int $messageCount): int
    {
        // TODO: Implement consume() method.
    }
} 
```

# To run the tests using phpunit.
With Coverage: `php vendor/bin/phpunit --coverage-text`
Without Coverage: `php vendor/bin/phpunit`

**Comment by ins0** [comment]
ping @beyerz have a look

**Pull request merged**